src/EnergyPlus/MixedAir.cc

```diff
@@ -1050,8 +1050,6 @@ void GetOutsideAirSysInputs(EnergyPlusData &state)
                        OASys.ComponentTypeEnum(CompNum) == SimAirServingZones::CompType::UnitarySystemModel ||
                        OASys.ComponentTypeEnum(CompNum) == SimAirServingZones::CompType::DXSystem) {
                 OASys.ComponentIndex(CompNum) = CompNum;
-                OASys.compPointer[CompNum] =
-                    UnitarySystems::UnitarySys::factory(state, DataHVACGlobals::UnitarySys_AnyCoilType, OASys.ComponentName(CompNum), false, 0);
```

Comment:
This cause a recursive loop. This factory call to UnitarySystem, to manage the CoilSystem object, which had a HXAssisted object as a child, called the UnitarySystem, which called the CoilSystem, which called the HXAssisted, which requested the controller name and index for the water coil, which called the OAControllers, which called the OASystem to wind up right back here. Because the OASystem getInput had not completed yet and state.dataMixedAir->GetOASysInputFlag was still true on subsequent mining calls within this call stack decsribed here. The fix was to move the call to the factory below after where state.dataMixedAir->GetOASysInputFlag = false. There may be a better fix for this but this seems to work as expected. As more components get scooped up into UnitarySystem I think we should keep an eye out for this type of recursive calling.

    if (state.dataMixedAir->GetOASysInputFlag) {
        GetOutsideAirSysInputs(state);
        state.dataMixedAir->GetOASysInputFlag = false;
    }